<h1>Palmer Penguin Data Analytics</h1>
<hr>
<h3>Analyst: Ashlei Luiez Dando</h3>

In [89]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Importing Datasets</p>
<p>Converting Data Sets to Dataframe</p>
<ul><b>Requirements</ul>
<li>Name of Dataframe</li>
<li>Source</li>

In [90]:
penguins = pd.read_csv('Datasets//penguins.csv')

In [91]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3> Descriptive Analytics </h3>

In [92]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [93]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Cleaning the Data Frame</h3>

In [94]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>Selecting a column</h2>
<h3>Syntax</h3><br>
data_frame['column_name']

In [95]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [96]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [97]:
chinstrap_df = penguins[penguins['species'] == "Chinstrap"]

In [98]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [99]:
adelie_df = penguins[penguins['species'] == "Adelie"]

In [100]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [101]:
gentoo_df = penguins[penguins['species'] == "Gentoo"]

In [102]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [103]:
chinstrap_heavy = chinstrap_df[chinstrap_df['body_mass_g'] >= 4000]

In [104]:
chinstrap_heavy

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [105]:
chinstrap_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


In [106]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [107]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [108]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>HOW TO CHANGE DATA TYPE COLUMN</h2>

In [109]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [110]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [111]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h1>CREATING A NEW COLUMN</h1>

In [112]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [113]:
penguins['body_mass_kg'] = round(penguins['body_mass_g']/1000,1)

In [114]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year', 'body_mass_kg'],
      dtype='object')

<h1>EXPLORATORY DATA ANALYTICS</h1>

In [115]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [116]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [117]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

<h1>1. Which female and male species had the longest average flippers recorded on Biscoe Island in 2009?</h1>

In [118]:
flipperLengthMale = penguins[(penguins['island'] == 'Biscoe') & (penguins['sex'] == 'male') & (penguins['year'] == 2009)]
flipperLengthMale.value_counts().sum()

np.int64(32)

In [119]:
flipperLengthFemale = penguins[(penguins['island'] == 'Biscoe') & (penguins['sex'] == 'female') & (penguins['year'] == 2009)]
flipperLengthFemale.value_counts().sum()

np.int64(28)

In [120]:
np.round(flipperLengthMale[flipperLengthMale['species'] == 'Gentoo']['flipper_length_mm'].mean())

np.float64(222.0)

In [121]:
np.round(flipperLengthMale[flipperLengthMale['species'] == 'Adelie']['flipper_length_mm'].mean())

np.float64(194.0)

In [122]:
np.round(flipperLengthMale[flipperLengthMale['species'] == 'Chinstrap']['flipper_length_mm'].mean())

np.float64(nan)

<h2>#1 INSIGHT</h2>

<h3>Male Gentoo penguins have an average flipper length of 218mm, making them the longest flippers recorded on Biscoe Island in 2009, while no female penguins were analyzed.</h3>

<hr>

<h1>2. Between 2007 and 2009, which year had the most penguins recorded?</h1>

In [123]:
penguins[penguins['year'] == 2007].value_counts().sum()

np.int64(110)

In [124]:
penguins[penguins['year'] == 2009].value_counts().sum()

np.int64(120)

<h2>#2 INSIGHT</h2>

<h3>2009 had the highest number of penguins recorded, with 120 counted, while 2007 had only 110.</h3>

<hr>

<h1>3. What island had the highest number of male penguins with a body mass over 4000g?</h1>

In [125]:
penguins[(penguins['sex'] == 'male') & (penguins['body_mass_g'] >= 4000)]['island'].value_counts()

island
Biscoe       77
Dream        29
Torgersen    15
Name: count, dtype: int64

<h2>#3 INSIGHT</h2>

<h3>Biscoe Island had the highest number of male penguins with a body mass over 4000g.</h3>

<hr>

<h1>4. What penguin species has a bill depth of less than 15mm?</h1>

In [126]:
penguins[penguins['bill_depth_mm'] <= 15]['species'].value_counts()

species
Gentoo       70
Adelie        0
Chinstrap     0
Name: count, dtype: int64

<h2>#4 INSIGHT</h2>

<h3>Gentoo species has a bill depth of less than 15mm while other species don't have bill depth of less than 15mm.</h3>

<hr>

<h1>5. What species had the highest count of living penguins on Torgersen Island recorded in 2007? </h1>

In [127]:
countsOfLiving = penguins[(penguins['island'] == 'Torgersen') & (penguins['year'] == 2007)]
countsOfLiving['species'].value_counts()

species
Adelie       20
Chinstrap     0
Gentoo        0
Name: count, dtype: int64

<h2>#5 INSIGHT</h2>

<h3>Torgersen Island had more Adelie penguins in the 2007 record.</h3>

<hr>

<h1>6. What island had the most living female penguins recorded in 2008?</h1>

In [128]:
count1 = penguins[(penguins['island'] == 'Biscoe') & (penguins['year'] == 2008)]
count1[count1['sex'] == 'male'].value_counts().sum()

np.int64(33)

In [129]:
count1 = penguins[(penguins['island'] == 'Dream') & (penguins['year'] == 2008)]
count1[count1['sex'] == 'male'].value_counts().sum()

np.int64(17)

In [130]:
count1 = penguins[(penguins['island'] == 'Torgersen') & (penguins['year'] == 2008)]
count1[count1['sex'] == 'male'].value_counts().sum()

np.int64(8)

<h2>#6 INSIGHT</h2>

<h3>Biscoe Island had the most living female penguins recorded in 2008.</h3>

<hr>

<h1>7. In 2008 records, which penguin species had a bill length of less than 50 mm, a flipper length greater than 200 mm, and a bill depth of less than 18 mm?</h1>

In [131]:
records = penguins[(penguins['year'] == 2008) & (penguins['bill_length_mm'] <= 50) & (penguins['flipper_length_mm'] >= 200) & (penguins['bill_depth_mm'] <= 18)]
records[records['species'] == 'Adelie'].value_counts().sum()

np.int64(1)

In [132]:
records[records['species'] == 'Gentoo'].value_counts().sum()

np.int64(39)

In [133]:
records[records['species'] == 'Chinstrap'].value_counts().sum()

np.int64(0)

<h2>#7 INSIGHT</h2>

<h3>1 Adelie and 39 Gentoo penguins had a bill length of less than 50 mm, a flipper length greater than 200 mm, and a bill depth of less than 18 mm, while no Chinstrap penguins met these criteria.</h3>

<hr>

<h1>DATA AGGREGATION</h1>
<hr>

In [134]:
penguins.groupby(['species','island','year']).size()

species    island     year
Adelie     Biscoe     2007    10
                      2008    18
                      2009    16
           Dream      2007    20
                      2008    16
                      2009    20
           Torgersen  2007    20
                      2008    16
                      2009    16
Chinstrap  Biscoe     2007     0
                      2008     0
                      2009     0
           Dream      2007    26
                      2008    18
                      2009    24
           Torgersen  2007     0
                      2008     0
                      2009     0
Gentoo     Biscoe     2007    34
                      2008    46
                      2009    44
           Dream      2007     0
                      2008     0
                      2009     0
           Torgersen  2007     0
                      2008     0
                      2009     0
dtype: int64

In [135]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  body_mass_kg  
0    3750.000000    male  2007           3.8  
1    3800.000000  female  2007           3.8  
2    3250.000000  female  2007           3.2  
3    4201.754386    male  2007           4.2  
4    3450.000000  female  2007           3.4  
..           ...     ...   ...           ...  
339  4000.000000    male  2009           4.0  
340  3400.000000  female  2009           3.4  
341  3775.000000    male  2009           3.8  
342  4100.000000    male  2009           4.1  
343  3775.000000  female  2009           3.8  

[344 rows x 9 columns]

<h1>Pearson Correlation</h1>

In [136]:
penguins['bill_length_mm'].corr(penguins['bill_depth_mm'])

np.float64(-0.23505287035553282)

<h1>Numerical - SciPy(Scientific Python)</h1>
<h3>Library - SciPy</h3>

In [137]:
from scipy.stats import f_oneway

In [138]:
x_df = penguins[penguins['sex'] == 'male']

In [139]:
x_df['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [140]:
f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

F_onewayResult(statistic=np.float64(68.38476171974526), pvalue=np.float64(3.0226600487050246e-15))

In [141]:
result, pvalue = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

In [142]:
result

np.float64(68.38476171974526)

In [143]:
pvalue

np.float64(3.0226600487050246e-15)